overlord/configstate/configcore/watchdog_test.go

```diff
@@ -0,0 +1,115 @@
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type watchdogSuite struct {
+	configcoreSuite
+
+	mockEtcEnvironment string
+}
+
+var _ = Suite(&watchdogSuite{})
+
+func (s *watchdogSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+
+	dirs.SetRootDir(c.MkDir())
+	err := os.MkdirAll(filepath.Join(dirs.GlobalRootDir,
+		"/etc/systemd/system.conf.d/"), 0755)
+	c.Assert(err, IsNil)
+	s.mockEtcEnvironment = filepath.Join(dirs.GlobalRootDir,
+		"/etc/systemd/system.conf.d/10-ubuntu-core-watchdog.conf")
+}
+
+func (s *watchdogSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+func (s *watchdogSuite) TestConfigureWatchdog(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	for option, val := range map[string]string{"runtime-timeout": "10", "shutdown-timeout": "60"} {
+
+		err := configcore.Run(&mockConf{
+			state: s.state,
+			conf: map[string]interface{}{
+				fmt.Sprintf("watchdog.%s", option): val,
+			},
+		})
+		c.Assert(err, IsNil)
+
+		var systemdOption string
+		switch option {
+		case "runtime-timeout":
+			systemdOption = "RuntimeWatchdogSec"
+		case "shutdown-timeout":
+			systemdOption = "ShutdownWatchdogSec"
+		}
+		c.Check(s.mockEtcEnvironment, testutil.FileEquals,
+			fmt.Sprintf("[Manager]\n%s=%s\n", systemdOption, val))
+	}
+}
+
+func (s *watchdogSuite) TestConfigureWatchdogAll(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	times := []int{10, 100}
+	err := configcore.Run(&mockConf{
+		state: s.state,
+		conf: map[string]interface{}{
+			"watchdog.runtime-timeout":  times[0],
+			"watchdog.shutdown-timeout": times[1],
+		},
+	})
+	c.Assert(err, IsNil)
+	c.Check(s.mockEtcEnvironment, testutil.FileContains, "[Manager]\n")
+	c.Check(s.mockEtcEnvironment, testutil.FileContains,
+		fmt.Sprintf("RuntimeWatchdogSec=%d\n", times[0]))
+	c.Check(s.mockEtcEnvironment, testutil.FileContains,
+		fmt.Sprintf("ShutdownWatchdogSec=%d\n", times[1]))
+}
+
+func (s *watchdogSuite) TestConfigureWatchdogBadFormat(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	err := configcore.Run(&mockConf{
+		state: s.state,
+		conf: map[string]interface{}{
+			"watchdog.runtime-timeout": "BAD",
```

Comment:
Might be nice to add a test here with an invalid duration, something like `20x` or so